r4/ccdatofhir-service/service.bal

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerinax/health.fhir.r4;
+import ballerinax/health.fhir.r4utils.ccdatofhir;
+
+# This service supports transform CCDA documents to FHIR based on the CCDA to FHIR mapping Implementation Guide.
+# Link to the IG: http://hl7.org/fhir/us/ccda/2023May/CF-index.html
+# The service is exposed at `/transform` path and the service is listening to HTTP requests at port `9090`.
+service / on new http:Listener(9090) {
+
+    # CCDA to FHIR transform service
+    # + return - Transformed FHIR bundle for the given CCDA document.
+    resource function post transform(http:RequestContext ctx, http:Request request) returns http:Response {
+
+        xml|error xmlPayload = request.getXmlPayload();
+        http:Response response = new;
+        if xmlPayload is error {
+            response.statusCode = http:STATUS_BAD_REQUEST;
+            r4:OperationOutcome operationOutcome = r4:errorToOperationOutcome(r4:createFHIRError(
+                "Invalid xml document.", r4:CODE_SEVERITY_ERROR,r4:TRANSIENT_EXCEPTION, cause = xmlPayload));
```

Comment:
Don't we have to extract the cause from the `xmlPayload`?

Reply:
Can you share an error generated by this?

Reply:
> {"resourceType":"OperationOutcome","meta":{"profile":["http://hl7.org/fhir/StructureDefinition/OperationOutcome"]},"issue":[{"severity":"error","diagnostics":"01ee3d7b-d8a4-1936-a24a-a3ec8c7b800a","code":"exception","details":{"coding":[{"system":"http://hl7.org/fhir/issue-type","code":"500"}],"text":"Invalid xml document."}}]}

Seems cause details are not picked from the OpOutcome. will check

Reply:
Cause details are picked with diagnostic field when creating OperationOutcome. Now the operationoutcome will look like following.

```
{
    "resourceType": "OperationOutcome",
    "meta": {
        "profile": [
            "http://hl7.org/fhir/StructureDefinition/OperationOutcome"
        ]
    },
    "issue": [
        {
            "severity": "error",
            "diagnostics": "01ee3d93-7657-18ce-a135-23cabd9fad70 : Error occurred while extracting xml data from entity: error(\"failed to parse xml: Unexpected end of input block in end tag\n at [row,col {unknown-source}]: [4,18]\")",
            "code": "exception",
            "details": {
                "coding": [
                    {
                        "system": "http://hl7.org/fhir/issue-type",
                        "code": "500"
                    }
                ],
                "text": "Invalid xml document. due to Error occurred while extracting xml data from entity: error(\"failed to parse xml: Unexpected end of input block in end tag\n at [row,col {unknown-source}]: [4,18]\")"
            }
        }
    ]
}
```